### PR TITLE
test: add test script

### PR DESCRIPTION
**Pull request by PhilipTamb**
This pull request wants to add the possibility to test a specific Zenoh-Flow configuration using a script

### CODE REVIEW

File: manual-tests/README.md [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+The script test verifies:
+- Zenoh with zenoh-flow plugin is up and running.
+- The example runs correctly.
+- That the number of active Zenoh-Flow nodes is equal to the number of nodes declared into the yaml file.
 
 ## How to run
 To do this you need to run :
 ```bash
-./automated-test.py
+pip install psutil
```
Comment (gabrik) [inline manual-tests/README.md:19]: Why not create a `requirements.txt` and install it with pip?
`pip3 install -r requirements.txt`

Also the version of the dependencies should be explicit, for reproducibility.

File: manual-tests/automated-test.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+import signal
+import  fnmatch
+import psutil
+import yaml
+from yaml.loader import SafeLoader
+
+
+class ZF_conf:
+    def __init__(self):
+        self.zenohd_path = "none"
```
Comment (gabrik) [inline manual-tests/automated-test.py:16]: use `None` instead of the string.

File: manual-tests/automated-test.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  if platform.system() == "Windows":
+      lib_name = "libzenoh_plugin_zenoh_flow.dll"
+  elif platform.system() == "Linux":
+      lib_name = "libzenoh_plugin_zenoh_flow.so"
+  elif platform.system() == "Darwin":
+      lib_name = "libzenoh_plugin_zenoh_flow.dylib"
+
+  libzenoh_plugin_path = find(lib_name,zenoh_flow_path)
+  zf_conf.libzenoh_plugin = libzenoh_plugin_path[0]
+  print(libzenoh_plugin_path[0])
```
Comment (gabrik) [inline manual-tests/automated-test.py]: use `print(f'Path of plugin library: {libzenoh_plugin_path[0]'}`

File: manual-tests/automated-test.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+process_name = "zenohd"
+pid = None
+for process in psutil.process_iter():
+    if process_name in process.name():
+       pid = process.pid
+       print("[info] kill process:", pid)
+       os.kill(pid, signal.SIGKILL) 
+       break
+
+print("[info] Launching Zenod process:  ", zf_command['zenohd_cmd'])
```
Comment (gabrik) [inline manual-tests/automated-test.py]: use `print(f'.....')`

**Review by gabrik — CHANGES_REQUESTED** [review]
When printing use the formatter with `print(f'Some thing: {variable_name}')`
https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings

File: manual-tests/automated-test.py [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    description="Script to automatically test that zenoh-flow works"
+)
+
+parser.add_argument(
+    "-b",
+    "--build",
+    nargs="?",
+    type=str,
+    default="False",
+    help="Specifies a different directory for Zenoh-Flow artifacts (default=debug)",
```
Comment (J-Loudet) [inline manual-tests/automated-test.py:77]: Shouldn't the default be "debug"?

File: manual-tests/automated-test.py [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    or args.zfctl == "/.config/zenoh-flow/zfctl.json"
+):
+    print(home_path)
+    zf_conf.zenohd_path = home_path + str(args.zenohd)
+    zf_conf.zfctl_path = home_path + str(args.zfctl)
+    zf_conf.zf_plugin_path = home_path + str(args.plugin)
+else:
+    zf_conf.zenohd_path = args.zenohd
+    zf_conf.zfctl_path = args.zfctl
+    zf_conf.zf_plugin_path = args.plugin
```
Comment (J-Loudet) [inline manual-tests/automated-test.py:119]: It is better to put the correct path in the default options. If users run the "help" option and see `"/.config/…"` they can assume that something is wrong.

There is the function: `os.path.expanduser()`.

What I suggest you do: provide a default option with `~` in the path & call `expanduser` in all cases (be it the default option or provided by the user).

File: manual-tests/automated-test.py [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+        "ZFCTL_CFG="
+        + zf_conf.zfctl_path
+        + " "
+        + home_path
+        + "/.local/bin/zfctl"
+        + " list instances"
+    )
+    print(f'show list of Zenoh-Flow instances: \n {zf_command["instances_cmd"]}')
+
+print("[Info] Looking for Zenohd process...")
```
Comment (J-Loudet) [inline manual-tests/automated-test.py:225]: Given that you are not just looking for `zenohd` processes, I would rather write:
```suggestion
print("[Info] Killing extra `zenohd` processes...")
```

File: manual-tests/automated-test.py [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+process_name = "zenohd"
+pid = None
+for process in psutil.process_iter():
+    if process_name in process.name():
+        pid = process.pid
+        print(f"[Info] kill process: {pid}")
+        os.kill(pid, signal.SIGKILL)
```
Comment (J-Loudet) [inline manual-tests/automated-test.py:232]: Couldn't this be simplified like so?
```suggestion
        print(f"[Info] kill process: {process.pid}")
        process.kill()
```

File: manual-tests/automated-test.py [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+    if str(tuple(data["operators"]).__len__()) == instance["operators"]:
+        print(
+            f'[Info] test active Operator nodes: SUCCESS, the active Operator nodes [{str(tuple(data["operators"]).__len__())}] are equal declared Operator nodes [{instance["operators"]}]'
+        )
+    if (
+        str(tuple(data["sources"]).__len__()) != instance["sources"]
+        or str(tuple(data["sinks"]).__len__()) != instance["sinks"]
+        or str(tuple(data["operators"]).__len__()) != instance["operators"]
+    ):
+        print("[Error] Test Zenoh-Flow configuration nodesSome nodes:  FAILED.")
```
Comment (J-Loudet) [inline manual-tests/automated-test.py:317]: Considering the previous change I asked you to do, these lines need to be removed as well.

**Review by J-Loudet — CHANGES_REQUESTED** [review]
Solid basis. Needs adjustments.

**Pull request closed without merging**